### ISSUE TITLE
Failing to send emails using Mailcow SMTP server

### ISSUE THREAD

**Issue by JacksonZ03** [supabase/supabase#21011]
# Bug report

I am running my own dockerised Mailcow SMTP server on an Ubuntu server.
I am able to send emails from the Sogo webclient interface and I have also tested sending emails using `swaks` as well.

### This is the swaks command I ran to send a test email using the comand line and it worked fine:
```
swaks --to redactedrecipient@gmail.com --from companyname-noreply@email.companyname.com --server mail.companyname.com -p 465 --auth LOGIN --auth-user companyname-noreply@email.companyname.com --auth-password 'redactedpassword' --header "From: Company Name <companyname-noreply@email.companyname.com>" --header "Subject: Test Email" --body "This is a test email to verify delivery." --tls-on-connect
```

### However, when I inputted the same details into Supabase like this:
<img width="1474" alt="image" src="https://github.com/supabase/supabase/assets/60581068/9d02b994-d4dc-4baa-a2cc-48cd7735f980">

### ...and tried to send a password recovery email as a test, I get this message:
<img width="374" alt="image" src="https://github.com/supabase/supabase/assets/60581068/44da99e0-70ed-45f6-b5d3-b7a50ad846d3">

I tried sending an email invite link and that gave me the same error on the top right hand corner too. 

## Expected Behaviour

A password reset email sent successfully to the recipient address.

## System information

- Server OS:  Ubuntu
- Browser: Chrome
- Web Supabase

**Comment by encima** [comment]
Is your Supabase also self-hosted or the Cloud version? Have you set the hostname for external availability or in `/etc/hosts`?
Are other SMTP services able to connect to Mailcow?

**Comment by JacksonZ03** [comment]
I'm using the cloud version on the supabase website. I am planning to self host, but I wanted to test to make sure it was working on the cloud version first.
I have only tested with swaks on multiple devices with different external IP's so I know it can be accessed remotely and I have also tried sending and receiving and both work.

The hostname is available externally and I made sure was set within the dockerised mailcow configuration file (mailcow.conf):
<img width="888" alt="image" src="https://github.com/supabase/supabase/assets/60581068/ea825d6c-7559-4bcc-a68a-d10a53000522">

**Comment by encima** [comment]
@JacksonZ03 

Hmm, this is odd. Would you be able to open a support ticket for this and include:
1. The link to this Issue
2. Your project ref
3. Mention Chris G (me) on the Success team and that I asked for this to be assigned to me
4. Confirmation to access your project using our internal tools

This allows us to check more deeply what is going on and I will update this issue with the cause when we know what it is.

**Comment by JacksonZ03** [comment]
Actually, I figured out what the problem was. After you mentioned testing with other SMTP clients, I went and set up keycloak to see if it would be able to send a test email...and it didn't.

So after I reviewed the logs, I realised the problem was with MailCow itself not having a valid SSL certificate files (`cert.pem` and `key.pem`) in the directory `~/mailcow-dockerised/data/assets/ssl/`

The reason I was able to send and receive from the sogo webclient was because I had the mailcow-webclient container routed through an nginx reverse proxy to expose it to port 80. But since the nginx reverse proxy used a local acme client which stored stored ssl certificates locally as well, non of the dockerised mailcow containers (dovecot, postfix, etc) could access it.

I'm still not quite sure how swaks was able to send emails with it even though it didn't work with regular smtp clients, but now it's fixed.
